compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -798,12 +806,30 @@ fn let_source(tcx: TyCtxt<'_>, pat_id: HirId) -> LetSource {
         ..
     }) = parent_parent_parent_parent_node
     {
-        LetSource::WhileLet
-    } else if let hir::Node::Expr(hir::Expr { kind: hir::ExprKind::If { .. }, .. }) =
-        parent_parent_node
+        return LetSource::WhileLet;
+    }
+
+    if let hir::Node::Expr(hir::Expr { kind: hir::ExprKind::If(..), .. }) = parent_parent_node {
+        return LetSource::IfLet;
+    }
+
+    LetSource::GenericLet
+}
+
+fn is_let_chain(tcx: TyCtxt<'_>, pat_id: HirId) -> bool {
+    let hir = tcx.hir();
```

Comment:
I'm not sure that this works for let chains with more than 1 `&&`. It might be simpler to check for the case this is not a let chain.

Reply:
It is working now for chains with more than one `&&` -> https://github.com/rust-lang/rust/pull/88642/files#diff-af0d549a94f69a8f6068e01f3df0cfad0bdcc2868bda71c416df6294abb241e6R820-R822